contracts/oracle/src/contract/alarms.rs

```diff
@@ -82,40 +76,44 @@ impl MarketAlarms {
             &below,
             AddAlarms {
                 storage,
-                addr,
+                receiver,
                 above,
                 price_alarms: Self::PRICE_ALARMS,
             },
         )
     }
 
-    pub fn try_notify_alarms<BaseC>(
-        storage: &mut dyn Storage,
-        mut batch: Batch,
-        prices: &[SpotPrice],
+    pub fn try_notify_alarms<'a, BaseC>(
+        &mut self,
+        storage: &dyn Storage,
+        prices: impl Iterator<Item = BasePrice<SwapGroup, BaseC>> + 'a,
         max_count: u32, // TODO: type alias
-    ) -> Result<Response, ContractError>
+    ) -> Result<Batch, ContractError>
```

Comment:
From a design perspective and for better testability we shall return here `impl Iterator<Item = Addr>`.